joint_state_controller:
  type: joint_state_controller/JointStateController
  publish_rate: 50

steer_wheel_controller:
  type: effort_controllers/JointTrajectoryController
  joints:
    - steering_1_joint
    - steering_2_joint
    - steering_3_joint
    - steering_4_joint
  gains:
    steering_1_joint: {p: 100,  d: 0, i: 0, i_clamp: 1}
    steering_2_joint: {p: 100,  d: 0, i: 0, i_clamp: 1}
    steering_3_joint: {p: 100,  d: 0, i: 0, i_clamp: 1}
    steering_4_joint: {p: 100,  d: 0, i: 0, i_clamp: 1}


leg_controller:
  type: effort_controllers/JointTrajectoryController
  joints:
    - leg_1_joint
    - leg_2_joint
    - leg_3_joint
    - leg_4_joint
  gains:
    leg_1_joint: {p: 100000,  d: 10, i: 0, i_clamp: 1}
    leg_2_joint: {p: 100000,  d: 10, i: 0, i_clamp: 1}
    leg_3_joint: {p: 100000,  d: 10, i: 0, i_clamp: 1}
    leg_4_joint: {p: 100000,  d: 10, i: 0, i_clamp: 1}

feet_controller:
  type: effort_controllers/JointTrajectoryController
  joints:
    - foot_1_joint
    - foot_2_joint
    - foot_3_joint
    - foot_4_joint
  gains:
    foot_1_joint: {p: 20000,  d: 10, i: 0, i_clamp: 1}
    foot_2_joint: {p: 20000,  d: 10, i: 0, i_clamp: 1}
    foot_3_joint: {p: 20000,  d: 10, i: 0, i_clamp: 1}
    foot_4_joint: {p: 20000,  d: 10, i: 0, i_clamp: 1}

wheel_1_velocity_controller:
  type: effort_controllers/JointVelocityController
  joint: wheel_1_joint
  pid:
    p: 1
    i: 0
    d: 0
    i_clamp: 1
wheel_2_velocity_controller:
  type: effort_controllers/JointVelocityController
  joint: wheel_2_joint
  pid:
    p: 1
    i: 0
    d: 0
    i_clamp: 1

wheel_3_velocity_controller:
  type: effort_controllers/JointVelocityController
  joint: wheel_3_joint
  pid:
    p: 1
    i: 0
    d: 0
    i_clamp: 1

wheel_4_velocity_controller:
  type: effort_controllers/JointVelocityController
  joint: wheel_4_joint
  pid:
    p: 1
    i: 0
    d: 0
    i_clamp: 1